app/lib/feed_manager.rb

```diff
@@ -424,24 +424,12 @@ def filter_from_list?(status, list)
   # @param [Symbol] context
   # @return [Boolean]
   def phrase_filtered?(status, receiver_id, context)
-    active_filters = Rails.cache.fetch("filters:#{receiver_id}") { CustomFilter.where(account_id: receiver_id).active_irreversible.to_a }.to_a
+    active_filters = CustomFilter.cached_filters_for(receiver_id)
+    active_keywords = active_filters.filter_map { |custom_filter, rules| rules[:keywords] if custom_filter.context.include?(context.to_s) && custom_filter.irreversible? }
```

Comment:
Irreversability of filters still makes me uneasy as this is basically an entirely different code path for filtering stuff to how it's done otherwise. I wonder if it would not be, after all, better to rely entirely on clients hiding items. I know that I initially came up with the `irreversible` flag with the reasoning being to ensure correct pagination (not have to deal with "empty" pages being returned) but I think it might be worth it to have a more straightforward filtering path.

Reply:
It's actually a leftover from the early rewriting before I had updated the API everywhere. I intend to remove this to not make it “irreversible”.

Reply:
Removed.